File: Frontend/src/types/employee.ts
export interface EmployeeResponse {
    id: string;
    firstName: string;
    lastName: string;
    middleName: string | null;
    email: string;
}

export interface CreateEmployeeRequest {
    firstName: string;
    lastName: string;
    middleName?: string;
    email: string;
}

export interface UpdateEmployeeRequest {
    firstName: string;
    lastName: string;
    middleName?: string;
    email: string;
}

export interface EmployeeShortResponse {
    id: string;
    fullName: string;
}

File: ProjectManagement.API/Domain/Entities/Employee.cs
namespace ProjectManagement.API.Domain.Entities;

public class Employee
{
    public Guid Id { get; set; } = Guid.CreateVersion7();

    public required string FirstName { get; set; }

    public required string LastName { get; set; }

    public string? MiddleName { get; set; }

    public required string Email { get; set; }

    public ICollection<Project> ManagedProjects { get; } = [];

    public ICollection<Project> Projects { get; } = [];
}